### PR TITLE
update button functionality and summary update

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
# PR 19-06-24
## Overview
In this PR we implement the functionality for the update button
## Key Changes :
 1) cart.js : 
    - Now once we modify anything on the cart and we update the prices the change will remain 
    - The summary section and the subtotal section automatically adapt to the changes

Animation

### CODE REVIEW

File: src/javaScript/cart.js [hunk 1]
```diff
@@ -101,25 +118,63 @@ function updateQuantityColor () {
...
-      quantity.textContent = Math.max(0, parseInt(quantity.textContent) - 1)
+      const index = subtractItem.dataset.index
+      const cartItems = getCartItems()
+      cartItems[index].quantity = Math.max(0, cartItems[index].quantity - 1)
+      setCartItems(cartItems)
+      quantities[index].textContent = cartItems[index].quantity
+      quantities[index].style.color = 'red'
+
+      const priceNumber = parseFloat(
+        cartItems[index].price.replace(/[^0-9.-]+/g, '')
```
Comment (vladiiancu) [inline src/javaScript/cart.js:154]: Why is this needed?

Comment (Alexandru-Bulai) [reply src/javaScript/cart.js:154]: To prevent any error when we convert strings to numbers, so by using `.replace` i ensure that only numeric characters, the decimal point, and the minus sign are kept
